niworkflows/utils/spaces.py

```diff
@@ -508,17 +508,55 @@ def insert(self, index, value, error=True):
         elif error is True:
             raise ValueError('space "%s" already in spaces.' % str(value))
 
-    def get_std_spaces(self, only_names=True, dim=(2, 3)):
-        """Return only standard spaces."""
-        names = []
-        std_spaces = []
-        for s in self._spaces:
-            if s.standard and s.dim in dim and s.fullname not in names:
-                names.append(s.fullname)
-                std_spaces.append(s)
-        if only_names:
-            return names
-        return std_spaces
+    def get_std_spaces(self, only_names=True, dim=(2, 3), cached=False):
+        """Return standard spaces.
+
+        Parameters
+        ----------
+        only_names : :obj:bool, optional
+            If `True` (default), return all unique :obj:Spaces names. If `False`,
+            return all `Space` entities.
+        dim : :obj:tuple, optional
+            Desired dimensions of the standard spaces (default is (2, 3))
+        cached : :obj:bool, optional
+            If `True`, query cached :obj:Spaces. If `False` (default), query all :obj:Spaces.
+
+        Examples
+        --------
+        >>> spaces = SpatialReferences(['MNI152NLin6Asym', ("fsaverage", {"den": "10k"})])
+        >>> spaces.get_std_spaces()
+        ['MNI152NLin6Asym', 'fsaverage']
+
+        >>> spaces.get_std_spaces(only_names=False)
+        [Space(name='MNI152NLin6Asym', spec={}),
+         Space(name='fsaverage', spec={'den': '10k'})]
+
+        >>> spaces.get_std_spaces(dim=(3,))
+        ['MNI152NLin6Asym']
+
+        >>> spaces.checkpoint()
+        >>> spaces.add(('MNI152NLin6Asym', {'res': '2'}))
+        >>> spaces.get_std_spaces()
+        ['MNI152NLin6Asym', 'fsaverage']
+
+        >>> spaces.get_std_spaces(only_names=False)
+        [Space(name='MNI152NLin6Asym', spec={}),
+         Space(name='fsaverage', spec={'den': '10k'}),
+         Space(name='MNI152NLin6Asym', spec={'res': '2'})]
```

Comment:
I think I prefer the former behavior, which would drop the second instance of ``MNI152NLin6Asym``

Reply:
I understand dropping a duplicate template when looking at `only_names`, but shouldn't we list different specifications of the same templates when looking at all components of the `Space`s? Otherwise, this method is a bit deceiving IMO

Reply:
I guess the problem roots from the name we chose for the `Space` data type. Because name and spec pairs are not just spaces, they are spatial references. A space does not have any notion of the sampling grid. Actually, my original implementation did not have `only_names` flag.

Reply:
ok - three ways to solve this:

1) allow `only_names` to view all `Space` objects, regardless of duplicate templates. This will lighten the load of other queries (ex https://github.com/poldracklab/fmriprep/pull/1955/files#diff-928c5651a2bdc224e2a84acff159a0bcR194)
2) revert this to only list the first `Space`'s specs and send the other(s) into limbo.
3) remove `only_names` and just return template names

anything but 2 is fine with me.

Reply:
I'd go with 3 (remove `only_names`), returning unique names (in the order of first occurrence).

Reply:
For clarity/consistency, we might want to change https://github.com/poldracklab/niworkflows/pull/455/files#diff-fb99c1c1fd9dceadc84eb083f28aede6R564 to:
```Python
def get_nonstd_references(self, only_names=True, dim=(2, 3)):
```
or just
```Python
def get_nonstandard(self, only_names=True, dim=(2, 3)):
```

Reply:
sure, but then let's just drop only_names altogether

Reply:
And also, see suggestion on line 559